test/reline/test_face.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+require_relative 'helper'
+
+class Reline::Face::Test < Reline::TestCase
+  RESET_SGR = "\e[0m"
+
+  class WithInsufficientSetupTest < self
+    def setup
+      Reline::Face.config(:my_insufficient_config) do |face|
```

Comment:
For tests that call `Reline::Face.config`, should we also set its `@configs` to the Reline default in teardown?

Once merged, these tests will be ran on Ruby CI with other gems' tests in the same process. So if `Reline::Face.configs` change over tests, it may cause other tests to fail. I broke Ruby CI many times for things like this 😅

Reply:
Existing tests don't override configs of `:default` and `:completion_dialog` which are possible to be used by other tests, so I don't think Ruby CI will break without teardown...maybe 

On the other hand, it is also good to make the teardown process in advance I carelessly write a test that overrides the default config in the future.
Will it look like this?:

```ruby
# lib/reline/face.rb
class Reline::Face
  def self.reset_config
    @configs = {}
    Reline::Face.config(:default) do |face|
      face.define :default, style: :reset
      face.define :enhanced, style: :reset
      face.define :scrollbar, style: :reset
    end
    Reline::Face.config(:completion_dialog) do |face|
      face.define :default, foreground: :white, background: :cyan
      face.define :enhanced, foreground: :white, background: :magenta
      face.define :scrollbar, foreground: :white, background: :cyan
    end
  end
end

# test/reline/test_face.rb
class Reline::Face::Test < Reline::TestCase
  def teardown
     Reline::Face.reset_config
  end
end
```

I can't say if I should do this now or not. What do you think?

Reply:
https://github.com/ruby/reline/pull/552/commits/23d2360657fc9ed428d7ed6ff9b0629c1472b8df

I separated the initializing method into `reset_to_initial_config` and `load_initial_config` so that a user can only reset the "initial" config (`:default` and `:completion_dialog`) without losing definitions by the user.

## initial vs default
I chose the word "initial" instead of "default" to avoid ambiguity because "default" is used elsewhere.

## config vs configs
In this case, no need for users to think about how many settings exist. So I think the singular "config" is enough and better.

Reply:
> I chose the word "initial" instead of "default" to avoid ambiguity because "default" is used elsewhere.

👍 

> In this case, no need for users to think about how many settings exist. So I think the singular "config" is enough and better.

I'm not sure if I agree. Using `config` means explicitly "there's only 1 config", which is simply incorrect in this case. But using `configs` actually could mean 1 or more.

Reply:
I still don't get it. `config` is never uncountable. Otherwise, we wouldn't have `Reline::Face.configs`, no? 

Even if we do want to consider it uncountable for this API, it's more confusing to have both countable and uncountable form of the same word in the same project, especially on the same component:

```rb
Reline::Face.load_initial_config #=> looks like it's loading a single config
Reline::Face.configs #=> returns 2 values?
```

The fact is that we do load 2 configs when calling that method, and we do call `configs` when retrieving them. Why insisting it being called either singular or uncountable?

Reply:
I see, fixed https://github.com/ruby/reline/pull/552/commits/7c5133c7b2b90cb37d92f667f361002af2302b94